modules/imap/functions.php

```diff
@@ -1523,9 +1523,13 @@ function parse_sieve_config_host($host) {
     $url = parse_url($host);
     if(!isset($url['host'])) {
         $host = $url['path'];
+    } else {
+        $host = $url['host'];
     }
     $port = $url['port'] ?? '4190';
-    return [$host, $port];
+    $scheme = $url['scheme'] ?? 'tcp://';
```

Comment:
I'm not sure this will work with stalwart too, try and ping me

Reply:
I'll get back to you after the test

Reply:
https://github.com/cypht-org/cypht/issues/1153